### PR TITLE
Add typings for the Keycloak JS client

### PR DESCRIPTION
**Pull request by eppsilon**
- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present.

- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: keycloak-js/keycloak-js-tests.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import { Keycloak } from 'keycloak-js';
```
Comment (andy-ms) [inline keycloak-js/keycloak-js-tests.ts]: When I use the npm package like this:
```js
global.window = global;
console.log(require("keycloak"), require("keycloak").Keycloak);
```
I get `[Function: t] undefined`. So declaring a `Keycloak` export doesn't seem right.
Seems like [`export =`](http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html) would be more accurate.
As always, test it in your own code.

File: keycloak-js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,447 @@
+// Type definitions for keycloak-js 2.4
+// Project: https://github.com/keycloak/keycloak
```
Comment (andy-ms) [inline keycloak-js/index.d.ts]: Could you link to documentation about the JS client?

Comment (eppsilon) [inline keycloak-js/index.d.ts]: Here's the documentation: https://keycloak.gitbooks.io/securing-client-applications-guide/content/topics/oidc/javascript-adapter.html

I added links to the JSDoc comments on the constructor and the instance interface.

File: keycloak-js/keycloak-js-tests.ts (hunk 1 above)
Comment (eppsilon) [inline keycloak-js/keycloak-js-tests.ts]: You're right; sorry about that. I've fixed it (I hope) and updated the test file.

**Comment by askannon** [comment]
@eppsilon thanks for adding Keycloak JS. Would it be possible for you to also add the keycloak-authz.js types from the same NPM to it?

Thanks

**Pull request merged**